addons/product/models/product_template.py

```diff
@@ -928,7 +928,7 @@ def _get_possible_combinations(self, parent_combination=None, necessary_values=N
 
         necessary_values = necessary_values or self.env['product.template.attribute.value']
         necessary_attributes = necessary_values.mapped('attribute_id')
-        ptal_stack = [self.valid_product_template_attribute_line_ids.filtered(lambda ptal: ptal.attribute_id not in necessary_attributes)]
+        ptal_stack = [self.valid_product_template_attribute_line_ids.filtered(lambda ptal: ptal.attribute_id not in necessary_attributes) if necessary_values else self.valid_product_template_attribute_line_ids]
```

Comment:
So if I understand correctly this is just a perf improvement to avoid looping for no reason if `necessary_values` is empty?

The commit message should explain what it does better, including the perf gain if it is indeed a perf commit. Did you measure the impact? Considering `valid_product_template_attribute_line_ids` is typically not a big array, it doesn't seem very useful at first glance.

Reply:
I'll analyze the impact and drop/update the commit accordingly ;)